### PR TITLE
feat(KeyValue): keyValue component

### LINKED ISSUE #191: [Feat]:New Key Value component
**Description**: The `Key Value` component displays pairs of keys and their corresponding values, typically used for presenting data in a structured format. This component is ideal for use in settings pages, user profiles, or any context where clear label-value associations are necessary.

**Requirements**:

1. **Component Implementation:**

- Implement the `Key Value` component in TypeScript.
- Utilize SCSS with BEM methodology for styling.
- Integrate design tokens from @geneui/tokens for consistent colors, font sizes, spacing, and borders.
- Support different layouts (e.g., horizontal, vertical) and alignment options (e.g., left, center, right).

2. **Unit Tests:**

- Write unit tests using Jest and React Testing Library.
- Ensure test coverage for all props, including rendering of different layouts and alignment options.
- Test for edge cases such as missing keys or values and unsupported layout types.

3. **Storybook Stories:**

- Create a Storybook story for the Key Value component.
- Showcase different configurations (e.g., horizontal, vertical) and alignment styles.
- Create controls for the props so users can interact with the Key Value component in Storybook.
- Document all props of the public interfaces.

4. **Figma Design:**

Follow the design outlined in the provided [Figma link.](https://www.figma.com/design/XmW63z2fXfHBC95eCM36ed/Gene-UI%3A-Components?node-id=3825-23640&node-type=canvas&t=BW68yyejj2aaHpzs-0)

**Checklist**:

- [ ] Component is implemented in TypeScript.
- [ ] Styling follows SCSS using BEM methodology.
- [ ] Design tokens from @geneui/tokens are used for colors, font sizes, spacing, and borders.
- [ ] Unit tests written and cover different states and edge cases.
- [ ] Storybook stories created with examples of all props and configuration options.
- [ ] Figma design matches the component implementation.

### PR DESCRIPTION
**Pull request by arthur-grigoryan-softconstruct**
### Chromatic Build Summary 
**Build URL:** https://www.chromatic.com/build?appId=66daee7859a843dd87bfb3d7&number=311 
**Storybook URL:** https://66daee7859a843dd87bfb3d7-kbcfoboniw.chromatic.com/

### CODE REVIEW

File: src/components/molecules/KeyValue/KeyValue.tsx [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    /**
+     * Size
+     * Possible values: `medium | large`;
+     */
+    size?: "medium" | "large";
+    /**
+     * Adds space between key and value in horizontal direction.
+     */
+    spaceBetween?: boolean;
+    children: [ReactElement<IKeyProps>, ReactElement<IValueProps>];
```
Comment (GaroGabrielyan) [inline src/components/molecules/KeyValue/KeyValue.tsx:32]: Children prop must have comment (description)

File: src/components/molecules/KeyValue/Value.tsx [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import React, { FC, ReactElement } from "react";
+
+// Content
+import { IPillProps } from "../../atoms/Pill";
+import { ITextLinkProps } from "../../atoms/TextLink/TextLink";
+import { elementWithType } from "./KeyValue";
+
+export interface IValueProps {
+    size?: "large" | "medium";
```
Comment (GaroGabrielyan) [inline src/components/molecules/KeyValue/Value.tsx:9]: The props must have comments (descriptions).

File: src/components/molecules/KeyValue/KeyValue.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+import Value from "@components/molecules/KeyValue/Value";
+
+// Helpers
+import { args, propCategory } from "../../../../stories/assets/storybook.globals";
+
+const key = <Key infoText="Info text">Title</Key>;
+
+const meta: Meta<typeof KeyValue> = {
+    title: "Molecules/KeyValue",
+    component: KeyValue,
```
Comment (GaroGabrielyan) [inline src/components/molecules/KeyValue/KeyValue.stories.tsx:18]:
```suggestion
    component: KeyValue,
    subcomponents: { Value },
```

Add subcomponent to show documentation for Value component.

File: src/components/molecules/KeyValue/index.tsx [hunk 4]
```diff
@@ -0,0 +1 @@
+export { IKeyValueProps, default } from "./KeyValue";
```
Comment (GaroGabrielyan) [inline src/components/molecules/KeyValue/index.tsx:1]: Please fix exports.

File: src/index.ts [hunk 5]
```diff
@@ -17,6 +17,7 @@ export { default as Tooltip } from "./components/molecules/Tooltip";
 export { default as ProgressBar } from "./components/molecules/ProgressBar";
 export { Steps, Step, IStepProps, IStepsProps } from "./components/molecules/Steps";
 export { default as Tag, ITagProps } from "./components/molecules/Tag";
+export { default as KeyValue } from "./components/molecules/KeyValue";
```
Comment (GaroGabrielyan) [inline src/index.ts:20]: Please fix exports.

File: src/components/molecules/KeyValue/KeyValue.stories.tsx [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+        spaceBetween: false
+    } as IKeyValueProps
+};
+
+export default meta;
+
+type Story = StoryObj<IKeyValueProps>;
+
+export const Default: Story = {};
+Default.args = {} as IKeyValueProps;
```
Comment (GaroGabrielyan) [inline src/components/molecules/KeyValue/KeyValue.stories.tsx:39]: Please remove this unnecessary line
```suggestion
```

File: src/components/atoms/Info/Info.tsx [hunk 7]
```diff
@@ -70,6 +70,7 @@ const Info: FC<IInfoProps> = ({ infoText, disabled, size = "smallNudge", appeara
         <Tooltip text={infoText} alwaysShow={alwaysShow} appearance={appearance === "inverse" ? "inverse" : "default"}>
             <button
                 type="button"
+                aria-label="press enter to open tooltip"
```
Comment (GaroGabrielyan) [inline src/components/atoms/Info/Info.tsx:73]: The aria-label="press enter to open tooltip" could be improved for better accessibility and clarity. Here’s why:

1. Not all users rely on Enter: While Enter is commonly used for activation, users may also use Spacebar or other assistive technology interactions.
2. "Press enter" is instructional: ARIA labels should describe the element, not instruct interaction methods.
3. It’s not clear what "tooltip" refers to: A more descriptive label would help users understand the purpose of the button.


```suggestion
               aria-label="More information"
```

File: src/components/molecules/KeyValue/Key.tsx [hunk 8]
```diff
@@ -0,0 +1,24 @@
+import React, { FC } from "react";
+
+// Content
```
Comment (GaroGabrielyan) [inline src/components/molecules/KeyValue/Key.tsx:3]: What this comment mention?

File: src/components/molecules/KeyValue/KeyValue.stories.tsx [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+        children: [
+            key,
+            <Value>
+                <Pill text="Pill" isFill />
+            </Value>
+        ]
+    }
+};
+
+export const WithTextLinkValue: Story = {
```
Comment (GaroGabrielyan) [inline src/components/molecules/KeyValue/KeyValue.stories.tsx:52]: Rewrite all stories with render method to make code more readable.

```suggestion
export const WithTextLinkValue: Story = {
    render: (props) => {
        return (
            <KeyValue {...props}>
                <Key infoText="Info text">Title</Key>
                <Value>
                    <TextLink text="Text Link" href="" />
                </Value>
            </KeyValue>
        );
    }
};
```

File: src/components/molecules/KeyValue/KeyValue.tsx [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+}
+
+const pillSize: { [key: string]: IPillProps["size"] } = {
+    large: "medium",
+    medium: "small",
+    small: "smallNudge"
+};
+
+export const elementWithType = (
```
Comment (GaroGabrielyan) [inline src/components/molecules/KeyValue/KeyValue.tsx:41]: You can simplify this function.
```suggestion
export const elementWithType = (
    element: JSX.Element,
    type: string,
    keyValueSize: IKeyValueProps["size"] = "medium"
) => {
    const size = element.type.name === "Pill" ? pillSize[keyValueSize] : keyValueSize;
    return element?.type?.name === type ? { ...element, props: { ...element.props, size } } : null;
};
```
